tests/unit/_modules/test_cephprocesses.py

```diff
@@ -0,0 +1,506 @@
+        """
+        passing a osd_id of 1 to the process, it also mocks the 'open_files'
+        namedtuple to match that ID. This tests the regex.
+        """
+        proc = MockedPsUtil('ceph-osd', 0, 0, '/usr/bin/ceph-osd', osd_id=1)
+        procinfo = cephprocesses.ProcInfo(proc)
+        assert procinfo.osd_id == '1'
+
+    def test_map_open_proc_to_osd_id_2digits(self):
+        """
+        passing a osd_id of 20 to the process, it also mocks the 'open_files'
+        namedtuple to match that ID. This tests the regex.
+        """
+        proc = MockedPsUtil('ceph-osd', 0, 0, '/usr/bin/ceph-osd', osd_id=20)
+        procinfo = cephprocesses.ProcInfo(proc)
+        assert procinfo.osd_id == '20'
+
+    def test_map_open_proc_to_osd_id_multi_digits(self):
+        """
+        passing a osd_id of 99999 to the process, it also mocks the 'open_files'
+        namedtuple to match that ID. This tests the regex.
+        """
+        proc = MockedPsUtil('ceph-osd', 0, 0, '/usr/bin/ceph-osd', osd_id=99999)
+        procinfo = cephprocesses.ProcInfo(proc)
+        assert procinfo.osd_id == '99999'
+
+    def test_map_open_proc_to_osd_id_multi_raises(self):
+        """
+        passing a osd_id of None  to the process, it also mocks the 'open_files'
+        namedtuple to match that ID. This tests the Exception
+        """
+        proc = MockedPsUtil('ceph-osd', 0, 0, '/usr/bin/ceph-osd', osd_id=None)
+        with pytest.raises(cephprocesses.NoOSDIDFound):
+            procinfo = cephprocesses.ProcInfo(proc)
+
+class TestMetaCheck():
+
+    @pytest.fixture(scope='class')
+    def mc(self):
+        self.set_osd_list([])
+        yield cephprocesses.MetaCheck()
+
+    def set_osd_list(self, osd_list):
+        cephprocesses.__salt__ = {'osd.list': lambda : osd_list}
+
+    @pytest.mark.skip(reason='Not fully implemented yet')
+    def test_blacklist(self):
+        pass
+
+    @pytest.mark.skip(reason='Not fully implemented yet')
+    def test_expected_osds(self, mc):
+        """
+        If no blacklist is set, expected OSDs are the osds that
+        are returned from __salt__['osd.list']
+        """
+        expect = ['1', '2']
+        self.set_osd_list(expect)
+        assert mc.expected_osds == expect
+
+    @pytest.mark.skip(reason='Not fully implemented yet')
+    def test_expected_osds_blacklist(self, mc):
+        """
+        if OSDs are blacklisted, expect to see it removed
+        from the list
+        """
+        expect = ['1', '2', '3']
+        mc.blacklist = {'ceph-osd': ['2']}
+        self.set_osd_list(expect)
+        assert mc.expected_osds == ['1', '3']
+        
+    def mock_up(self):
+        ups = []
+        for prc_name, bin_names in cephprocesses.processes.items():
+            for bin_name in bin_names:
+                proc = MockedPsUtil(bin_name, 0, 0, '/usr/bin/{}'.format(bin_name))
+                procinfo = cephprocesses.ProcInfo(proc)
+                ups.append(procinfo)
+        return ups
+
```

Comment:
Perhaps you can use the setup and tear down concept of the pytest, ether on a per-module/class/function basis level.
Prime candidates for the setup would be mock_up, mc, psuil... May be others,
See more details in: https://docs.pytest.org/en/latest/xunit_setup.html?highlight=setup_class#method-and-function-level-setup-teardown

Reply:
I think I kinda do that on a class level rather than on a module/method level. Using plain old methods for this enables me to call those helpers whenever I need them. If I'm completely wrong, please enlighten me :)